### DISCUSSION TITLE
MongoDB Atlas connection failed in node.js app upon deploy

### DISCUSSION THREAD

**Discussion by snowmania96** [community/community#137390] [Programming Help] [answered]
### Body

I have recently deployed my node.js app on a hosting server and uses cpanel.
Once I run the server, it says that mongodb atlas refused connection, but I enabled 0.0.0.1(all connection from ip addresses) on network access section in my atlas account.
Is this because I have missed something in configuration?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Azymack** [comment, marked as answer]
I have encountered a similar issue before.
It's likely due to hosting provider's firewall configuration. (which blocks i/o request for a specific port. 27017 in your case)
Enable that port on WHM and let me know if connection still fails.
You can consult below links.

https://stackoverflow.com/questions/73150035/can-not-connect-to-mongodb-atlas-in-cpanel
https://www.eukhost.com/kb/how-to-open-ports-via-whm/

**Comment by Azymack** [comment]
Make sure you have root access.

**Comment by snowmania96** [comment]
Yes, I have.
I am checking it now.

**Comment by snowmania96** [comment]
Thanks, @Azymack.
You saved my day.
